main.go

```diff
@@ -161,7 +161,8 @@ func main() {
 	manageExistingClients()
 
 	// Now make sure that clients are in the appropriate visible state.
-	for _, wrk := range wm.Heads.Workspaces.Wrks {
+	for i, wrk := range wm.Heads.Workspaces.Wrks {
+		logger.Warning.Printf("Window %d is visible? %v", i, wrk.IsVisible())
```

Comment:
Might be better suited as `logger.Message`.

Reply:
oh dear, this needs to be gone, hang on ...